### ISSUE TITLE
'struct dirent' has no member named 'd_type' on Solaris

### ISSUE THREAD

**Issue by dago** [collectd/collectd#3194]
I am currently trying to compile the current collectd from master on Solaris 10 Sparc.
Among the PR #3193 the `struct dirent` does not have `d_type`:

```
  CC       src/utils/proc_pids/proc_pids_test.o
In file included from src/utils/proc_pids/proc_pids_test.c:1:0:
./src/utils/proc_pids/proc_pids.c: In function 'get_pid_number':
./src/utils/proc_pids/proc_pids.c:198:12: error: 'struct dirent' has no member named 'd_type'
   if (entry->d_type != DT_DIR)
            ^
./src/utils/proc_pids/proc_pids.c:198:24: error: 'DT_DIR' undeclared (first use in this function)
   if (entry->d_type != DT_DIR)
                        ^
./src/utils/proc_pids/proc_pids.c:198:24: note: each undeclared identifier is reported only once for each function it appears in
src/utils/proc_pids/proc_pids_test.c: In function 'stub_procfs_setup':
src/utils/proc_pids/proc_pids_test.c:43:45: error: format '%d' expects argument of type 'int', but argument 5 has type 'pid_t {aka const long int}' [-Werror=format=]
     snprintf(path, STATIC_ARRAY_SIZE(path), "%s/%d", proc_fs,
                                             ^
src/utils/proc_pids/proc_pids_test.c: In function 'test_get_pid_number__valid_dir':
src/utils/proc_pids/proc_pids_test.c:196:4: error: 'struct dirent' has no member named 'd_type'
   d.d_type = DT_DIR;
    ^
src/utils/proc_pids/proc_pids_test.c:196:14: error: 'DT_DIR' undeclared (first use in this function)
   d.d_type = DT_DIR;
              ^
src/utils/proc_pids/proc_pids_test.c: In function 'test_get_pid_number__invalid_dir_name':
src/utils/proc_pids/proc_pids_test.c:213:4: error: 'struct dirent' has no member named 'd_type'
   d.d_type = DT_DIR;
    ^
src/utils/proc_pids/proc_pids_test.c:213:14: error: 'DT_DIR' undeclared (first use in this function)
   d.d_type = DT_DIR;
              ^
src/utils/proc_pids/proc_pids_test.c: In function 'test_read_proc_name__valid_name':
src/utils/proc_pids/proc_pids_test.c:232:4: error: 'struct dirent' has no member named 'd_type'
   d.d_type = DT_DIR;
    ^
src/utils/proc_pids/proc_pids_test.c:232:14: error: 'DT_DIR' undeclared (first use in this function)
   d.d_type = DT_DIR;
              ^
src/utils/proc_pids/proc_pids_test.c: In function 'test_read_proc_name__invalid_name':
src/utils/proc_pids/proc_pids_test.c:250:4: error: 'struct dirent' has no member named 'd_type'
   d.d_type = DT_DIR;
    ^
src/utils/proc_pids/proc_pids_test.c:250:14: error: 'DT_DIR' undeclared (first use in this function)
   d.d_type = DT_DIR;
              ^
cc1: all warnings being treated as errors
```

As this is from the testsuite: should this test best be skipped after verification of absence of `d_type` or is the test worth to be rewritten?

**Comment by dago** [comment]
Looking in `configure.ac` it appears that `src/utils/proc_pids/proc_pids.c` should only be build if `BUILD_PLUGIN_INTEL_RDT` is set, which it is not on my Solaris box. Also inside the conditional `BUILD_PLUGIN_INTEL_RDT` there is
```
if BUILD_PLUGIN_INTEL_RDT
...
test_plugin_intel_rdt_SOURCES = \
        src/intel_rdt_test.c \
        src/utils/config_cores/config_cores.c \
        src/utils/proc_pids/proc_pids.c \
        src/daemon/configfile.c \
        src/daemon/types_list.c
...
check_PROGRAMS += test_plugin_intel_rdt
endif
```
But then the respective test for `proc_pids.c` is added unconditionally:
```
check_PROGRAMS = \
        test_common \
...
        test_utils_proc_pids
```

This results in a missing rule in the Makefile when `src/utils/proc_pids/proc_pids.c` is changed it is not being rebuild before the new test. Additionally, I wonder if `src/utils/proc_pids/proc_pids.c` should be used and tested even if `BUILD_PLUGIN_INTEL_RDT` is not set.

**Comment by dago** [comment]
I wonder if just `test_utils_proc_pids` should be removed from the unconditional list in `check_PROGRAMS` and added to `check_PROGRAMS` inside the conditional of `BUILD_PLUGIN_INTEL_RDT`.

Maybe someone can clarify? If the test should be in place I can prepare a patch for the test instead,

**Comment by rpv-tomsk** [comment]
Closed by  #3194.
